apps/sms/js/compose.js

```diff
@@ -835,34 +852,28 @@ var Compose = (function() {
         data: activityData
       });
 
-      activity.onsuccess = function() {
-        var result = activity.result;
+      return new Promise((resolve, reject) => {
```

Comment:
Since DOMRequest could return promise now, I think this change is reasonable and code would be more neat by using promise here. But this means the compose_test might also need to rewrite the mozActivity related part. Maybe we can stub mozactivity for this testing and leave the mock mozactivity inconsistancy until we rewrite all the mozactivity into promise, or we simply leave it as it is and rewrite all the mozactivity all at once including the test, wdyt?